provider/github/token_transport.go

```diff
@@ -70,13 +70,21 @@ func NewClientPoolFromTokens(
 		}
 
 		client := NewClient(rt, cache, conf.MinInterval, gitAuth, timeout)
+		if err := ValidateTokenPermissions(client); err != nil {
+			return nil, err
+		}
 
 		if _, ok := byClients[client]; !ok {
 			byClients[client] = []*repositoryInfo{}
 		}
 
 		byClients[client] = append(byClients[client], repos...)
 		for _, r := range repos {
+			err := ClientCanPush(client, r)
```

Comment:
This is actually checking if the token owner can post comments on the repo, right? It took me a moment, it looks like we will be _pushing_ code to the repo. Maybe it should be called `IsCollaborator`.

Reply:
It checks if the token owner can update status. It requires `push` access:

> Users with push access in a repository can create commit statuses for a given SHA.

https://developer.github.com/v3/repos/statuses/#create-a-status

`IsCollaborator` can't be used because collaborator can have only `pull` access. (oops, I don't check for it, I'll fix it).

How about `CanPostStatus`?

Reply:
much better, yes :+1: Otherwise it was a bit confusing

Reply:
fixed